Assets/UI/worldviewiconsmanager_CQUI.lua

```diff
@@ -7,27 +7,32 @@ include("WorldViewIconsManager");
 -- Cached Base Functions
 -- ===========================================================================
 BASE_SetResourceIcon = SetResourceIcon;
+BASE_AddImprovementRecommendationsForCity = AddImprovementRecommendationsForCity;
 
 -- ===========================================================================
 -- CQUI Members
 -- ===========================================================================
-local CQUI_ResourceIconStyle = 1;
+local CQUI_ResourceIconStyle:number = 1;
+local CQUI_ShowImprovementsRecommendations:boolean = false;
 
 function CQUI_GetSettingsValues()
-  CQUI_ResourceIconStyle = GameConfiguration.GetValue("CQUI_ResourceDimmingStyle");
-  if CQUI_ResourceIconStyle == nil then
-    print("CQUI_ResourceIconStyle is nil!  Using default value.");
-    CQUI_ResourceIconStyle = 1;
-  end
+	CQUI_ResourceIconStyle = GameConfiguration.GetValue("CQUI_ResourceDimmingStyle");
+	if CQUI_ResourceIconStyle == nil then
+		print("CQUI_ResourceIconStyle is nil!  Using default value.");
+		CQUI_ResourceIconStyle = 1;
+	end
+	CQUI_ShowImprovementsRecommendations = ( GameConfiguration.GetValue("CQUI_ShowImprovementsRecommendations") == 1);
 end
 
-function CQUI_OnIconStyleSettingsUpdate()
-  CQUI_GetSettingsValues();
-  Rebuild();
+function CQUI_OnSettingsUpdate()
+	CQUI_GetSettingsValues();
+	Rebuild();
+	if not CQUI_ShowImprovementsRecommendations then ClearImprovementRecommendations(); end -- allows to switch them off while the builder is still selected
```

Comment:
This will update the recommendations displayed when the setting is turned off while a builder is selected. Do you know if there is a way to make it work in the other direction? As in keep the setting disabled, then select builder, then turn on setting and show the recommendation on the map.